### ISSUE TITLE
Multiple EP attribution - result attribution only occurring on first provider

### ISSUE THREAD

**Issue by ewhitley** [pophealth/popHealth#186]
Hello - 

Wondering if this is by design, but figured I'd ask.  Using the multi-provider/site/facility support to attempt to attribute a single patient to multiple providers, but I'm not seeing that work as anticipated.
- Added a second provider to the example OPML file
  - Provider Jim Green NPI 1111 - nested under Org A
  - Provider Sally Smith NPI 2222 - nested under Org A
- Took sample Cypress QRDA I file and added two 
  <serviceEvent classCode="PCPR"> to include reference to both providers (NPI 1111 and NPI 2222)
- Ran numbers
- Expect IPP = 1, DENOM = 1, NUMER = 0

I'm seeing expected results for Org A and first provider, but not second provider.  Second provider gets IPP = 0, DENOM = 0.  I played with the OPML file and think the source is the measure calculations are only being done for the first provider listed.

Are you seeing this?  If so, is this by design?  Any option to have the ability to attribute a single patient to multiple providers?

Thank you,
Eric

**Comment by rdingwell** [comment]
Eric,

Do the record and patient_cache entries have the providers on them in the
db?

Rob

**Comment by ewhitley** [comment]
Hi, Rob - 
- patient_cache - I do see the providers.  All are appearing
- records -> provider_performances array - I do see the providers.  All are appearing

The providers do seem to be loading from QRDA I -> into the related patient records.  (Which was cool - I was happy to see that :)  ).

I noticed if I reorganize the OPML file and then re-import, it's always the first provider that calculates.  None of the child providers get results.  If I dig into the query_cache, I see the provider references, but all providers after the first have zeros.

Thank you,
Eric

Relevant bits of the OPML file (liberally swiped from the samples)

  &lt;outline id="1" VISN="1" name="VA New England Healthcare System" level="VISN"&gt;
    &lt;outline id="627" name="VA Connecticut Healthcare System, West Haven Campus" level="division" Division="627"&gt;
      &lt;outline id="6279AA" name="Newington VANURS" level="facility" Facility="6279AA"&gt;
          &lt;outline id="11111111" facility="9238429385" name="Jim Green" level="provider" tin="123456790" npi="11111111" organization="General Hospital" specialty="200000000X" /&gt;
          &lt;outline id="9238429385" facility="9238429385" name="Jane Simpson" level="provider" tin="123456790" npi="9238429385" organization="General Hospital" specialty="200000000X" /&gt;
          &lt;outline id="222222" facility="222222" name="Sally Smith" level="provider" tin="222222" npi="222222" organization="General Hospital" specialty="200000000X" /&gt;
      &lt;/outline&gt;
    &lt;/outline&gt;
  &lt;/outline&gt;

**Comment by rdingwell** [comment]
Provider calculations are simply filters on the patient cache so if one provider is rolling up correctly the others should as well, unless the ids do not line up.  Do the provider ids in the record, patient_cache and query_cache all point back to the correct providers?

**Comment by rdingwell** [comment]
I'll take a look in the morning to see if there is something off in QME when it comes to filtering sub levels of providers.

**Comment by ewhitley** [comment]
Hi, Rob - 

I take it back - you're 100% right.  I mis-read the object_ids.

I'm _not_ actually seeing the provider ID references in...
- patient_cache - Not seeing 2nd, 3rd, etc. providers
- records -> provider_performances array - Not seeing 2nd, 3rd, etc. providers

I'm only seeing the provider references in the query_cache for that patient - not the records or patient_cache.  That would explain things.

I think I see what's going on (at least the result) - it appears the be the importer not picking up the second serviceEvent entry.  I checked in the records collection and (after flushing patient_cache and query_cache), hand-entered the expected JSON element for the second provider.  From there, everything did work.  I think that's the culprit (assuming we want this behavior - I do, but I don't know about others).

Thank you,
Eric

**Comment by ewhitley** [comment]
OK, I sorted this out.   As usual, all my fault and popHealth works perfectly :)

Two things:

1) I messed up the DOC attribution XML in the QRDA I.  Yeah - that'll do it... I nested multiple serviceEvent entries under a single documentationOf, which is a no-no.  Just needed to put out separate documentationOf entries per provider and that did the trick

2) In the OPML file I just followed the sample and blindly ran into a problem of my own creation

For a provider under another outline node, I supplied a facility that overlapped with the facility of another provider.  While the NPI was unique, the overlapping facility must have been confused things.  

EX:
          &lt;outline id="11111111" facility="9238429385" name="Jim Green" level="provider" tin="123456790" npi="11111111" organization="General Hospital" specialty="200000000X" /&gt;
          &lt;outline id="9238429385" facility="9238429385" name="Jane Simpson" level="provider" tin="123456790" npi="9238429385" organization="General Hospital" specialty="200000000X" /&gt;

 facility="9238429385"  here was enough to do it

Sorry for wasting your time, Rob
